wrapanapi/msazure.py

```diff
@@ -337,40 +353,46 @@ def list_network(self):
     def disconnect(self):
         pass
 
-    def remove_nics_by_search(self, nic_template):
+    def remove_nics_by_search(self, nic_template, resource_group=None):
         """
         Used for clean_up jobs to remove NIC that are not attached to any test VM
```

Comment:
This is actually an API change, I'd like to see some more description in the docblock now that resource_group is an option, and this doesn't operate on all resource groups.

Reply:
that's true

Reply:
When there is an optional parameter, such as `resource_group=None`, we need to document what happens when it is left at it's default value. I.e. "when None (default), the instance's resource group is used instead" like in this case, or "all resource groups are searched / changes are applied to all resource groups" in potentially other cases... I don't know which one it is, until I look at the code. So yes, we need to document this case and the others as well.

Reply:
updated